src/main/java/org/elastos/hive/vendors/onedrive/OneDriveAuthHelper.java

```diff
@@ -65,11 +65,15 @@ public AuthToken getToken() {
 
 	@Override
 	public CompletableFuture<AuthToken> checkExpired(Callback<AuthToken> callback) {
-		if (token.isExpired())
+		if (token.isExpired()) {
 			return redeemToken(callback);
+		}
 
 		CompletableFuture<AuthToken> future = new CompletableFuture<AuthToken>();
-		callback.onSuccess(token);
+		if (callback != null) {
+		    callback.onSuccess(token);			
+		}
```

Comment:
Not necessary to check callback is null, because callback always valid with user customised callback or NullCallback.

Reply:
Ok, i'll add the codes to check.